feat(drag-drop): add service for attaching drag&drop to arbitrary DOM nodes

crisbeto opened the pull request:
* Adds the  `DragDrop` service that simplifies the construction logic for `DragRef` and `DropListRef` and allows consumers to attach drag&drop functionality to arbitrary DOM nodes, rather than having to go through the directives.
* Reworks `DragRef` and `DragDropRef` to make them easier to construct.
* Normalizes some of the instances where some parameters only accept an `ElementRef`, whereas others accept `ElementRef | HTMLElement`.

devversion commented on src/cdk/drag-drop/directives/drag.spec.ts:
```diff
@@ -21,6 +21,7 @@ import {
   createTouchEvent,
 } from '@angular/cdk/testing';
 import {Directionality} from '@angular/cdk/bidi';
+import {of as observbleOf} from 'rxjs';
```
Typo? `observableOf`

crisbeto replied:
Ops, thanks for catching it.

crisbeto commented:
@jelbourn I've addressed the feedback. Regarding docs, I was planning on adding some live examples in a follow-up. As for tests, I'm not sure whether it makes sense to test this service, because all it does is create the refs which is covered by the TS type checking already.

jelbourn commented:
It would be good to have at least one test that verifies the services can be injected and invoked correctly from a code coverage standpoint.

crisbeto commented:
@jelbourn this PR should be good to go now.
